docs/resources/aws_s3_bucket.md

```diff
@@ -0,0 +1,123 @@
+
+As of this writing, this resource only supports evaluating the first two, bucket ACLs and bucket policies.  Evaluating object ACLs introduces scalability concerns that are currently not well handled by the AWS API, and may be better handled via AWS mechanisms such as CloudTrail and Config.
+
+In particular, users of the `be_public` matcher should carefully examine the conditions under which the matcher will detect an insecure bucket.  See the `be_public` section under the Matchers section below.
+
+## Syntax
+
+An `aws_s3_bucket` resource block declares a bucket by name, and then lists tests to be performed.
+
+    describe aws_s3_bucket(bucket_name: 'test_bucket') do
+      it { should exist }
+      it { should_not be_public }
+    end
+
+    describe aws_s3_bucket('test_bucket') do
+      it { should exist }
+    end
+
+<br>
+
+## Examples
+
+The following examples show how to use this InSpec audit resource.
+
+### Test a bucket's bucket-level ACL
+
+    describe aws_s3_bucket('test_bucket') do
+      its('bucket_acl.count') { should eq 1 }
+    end
+
+### Check to see if a bucket has a bucket policy
+
+    describe aws_s3_bucket('test_bucket') do
+      its('bucket_policy') { should be_empty }
+    end
+
+### Check to see if a bucket appears to be exposed to the public
+
+    # See Limitations section above
+    describe aws_s3_bucket('test_bucket') do
+      it { should_not be_public }
+    end
+<br>
+
+## Supported Properties
+
+### region
+
+The `region` property identifies the AWS Region in which the S3 bucket is located.
+
+    describe aws_s3_bucket('test_bucket') do
+      # Check if the correct region is set
+      its('region') { should eq 'us-east-1' }
+    end
+
+## Unsupported Properties
+
+### bucket_acl
+
+The `bucket_acl` property is a low-level property that lists the individual Bucket ACL grants that are in effect on the bucket.  Other higher-level properties, such as be\_public, are more concise and easier to use.  You can use the `bucket_acl` property to investigate which grants are in efffect, causing be\_public to fail.
+
+The value of bucket_acl is an Array of simple objects.  Each object has a `permission` property and a `grantee` property.  The `permission` property will be a string such as 'READ', 'WRITE' etc (See the [AWS documentation](https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#get_bucket_acl-instance_method) for a full list).  The `grantee` property contains sub-properties, such as `type` and `uri`.
+
+    
+    bucket_acl = aws_s3_bucket('my-bucket')
+
+    # Look for grants to "AllUsers" (that is, the public)
+    all_users_grants = bucket_acl.select do |g|
+      g.grantee.type == 'Group' && g.grantee.uri =~ /AllUsers/
+    end
+
+    # Look for grants to "AuthenticatedUsers" (that is, any authenticated AWS user - nearly public)
+    auth_grants = bucket_acl.select do |g|
+      g.grantee.type == 'Group' && g.grantee.uri =~ /AuthenticatedUsers/
+    end
+
+### bucket_policy
+
```

Comment:
We recommend using a higher-level property, such as be\_public, which is concise and easier to implement.